buffer/src/main/java/io/netty/buffer/Unpooled.java

```diff
@@ -213,9 +213,16 @@ public static ByteBuf wrappedBuffer(ByteBuffer buffer) {
     /**
      * Creates a new buffer which wraps the specified memory address. If {@code doFree} is true the
      * memoryAddress will automatically be freed once the reference count of the {@link ByteBuf} reaches {@code 0}.
+     * <p>
+     * <strong>Note:</strong> This method is inherently unsafe, and may not be supported on all platforms.
      */
     public static ByteBuf wrappedBuffer(long memoryAddress, int size, boolean doFree) {
-        return new WrappedUnpooledUnsafeDirectByteBuf(ALLOC, memoryAddress, size, doFree);
+        ByteBuffer buffer = PlatformDependent.directBuffer(memoryAddress, size);
```

Comment:
This means it will fail on Java9+, correcT ?

Reply:
Yes, that seems to be correct.